packages/ui/components/SurveysList/components/SurveyMigrateForm.tsx

```diff
@@ -0,0 +1,159 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useState } from "react";
+import { useFieldArray, useForm } from "react-hook-form";
+import toast from "react-hot-toast";
+import { TProduct } from "@formbricks/types/product";
+import {
+  TSurvey,
+  TSurveyMigrateFormData,
+  ZSurveyMigrateFormValidation,
+} from "@formbricks/types/surveys/types";
+import { Button } from "../../Button";
+import { FormField, FormProvider } from "../../Form";
+import { Label } from "../../Label";
+import { RadioGroup, RadioGroupItem } from "../../RadioGroup";
+import { TooltipRenderer } from "../../Tooltip";
+import { migrateSurveyToOtherEnvironmentAction } from "../actions";
+
+export const SurveyMigrateForm = ({
+  defaultProducts,
+  survey,
+  onCancel,
+  setOpen,
+  onMigrated,
+}: {
+  defaultProducts: TProduct[];
+  survey: TSurvey;
+  onCancel: () => void;
+  setOpen: (value: boolean) => void;
+  onMigrated: (surveyId: string) => void;
+}) => {
+  const [selectedEnvironmentId, setSelectedEnvironmentId] = useState<string | null>(null);
+
+  const form = useForm<TSurveyMigrateFormData>({
+    resolver: zodResolver(ZSurveyMigrateFormValidation),
+    defaultValues: {
+      products: defaultProducts.map((product) => ({
+        product: product.id,
+        environments: [],
+      })),
+    },
+  });
+
+  const formFields = useFieldArray({
+    name: "products",
+    control: form.control,
+  });
+
+  const onSubmit = async () => {
+    if (!selectedEnvironmentId) {
+      toast.error("Please select an environment to migrate the survey to");
+      return;
+    }
+    try {
+      const result = await migrateSurveyToOtherEnvironmentAction({
+        surveyId: survey.id,
+        targetEnvironmentId: selectedEnvironmentId,
+      });
+      if (result?.serverError) {
+        throw new Error("Server Error while trying to migrate the survey.");
+      }
+      toast.success("Survey migrated successfully!");
+      onMigrated(survey.id);
+    } catch (error) {
+      toast.error("Failed to migrate survey");
+    } finally {
+      setOpen(false);
+    }
+  };
+
+  return (
+    <FormProvider {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="relative flex h-full w-full flex-col gap-8 overflow-y-auto bg-white p-4">
+        <div className="space-y-8 pb-12">
+          <RadioGroup>
+            {formFields.fields.map((field, productIndex) => {
+              const product = defaultProducts.find((product) => product.id === field.product);
+              const isDisabled = survey.type !== "link" && product?.config.channel !== survey.type;
```

Comment:
A product can have its channel as `null`, and any type of survey can be created inside that(we already have a lot of null channelled surveys, and a new one can be created through the formbricks CX product onboarding), for which selection shouldn't be disabled.
![image](https://github.com/user-attachments/assets/bc40b6e6-7024-4ee7-8c6e-f7b11e1744e2)

Reply:
@pandeymangg & @mattinannt: Can you please confirm that all types of environment now support all types of surveys? (since app & website surveys have been combined in https://github.com/formbricks/formbricks/pull/3314)

Reply:
> @pandeymangg & @mattinannt: Can you please confirm that all types of environment now support all types of surveys? (since app & website surveys have been combined in #3314)

@jonas-hoebenreich yes, users can now create any type of survey in any type of product channel, the survey types have also been reduced to just two: `link` and `app` surveys. The channels are still all three types (`website`, `app`, `link`) but there are no survey type restrictions now for any channel type